gitlab/cmd/controller/main.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2020 The TriggerMesh Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"os"
+
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"knative.dev/eventing-contrib/gitlab/pkg/apis"
+	"knative.dev/eventing-contrib/gitlab/pkg/controller"
+	"knative.dev/eventing-contrib/gitlab/pkg/webhook"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/signals"
+)
+
+func main() {
+	var metricsAddr string
```

Comment:
should we use `sharedmain`, like described here:
https://knative.dev/development/eventing/samples/writing-receive-adapter-source/03-controller/

or done in other sources ? 
e.g. https://github.com/knative/eventing-contrib/blob/master/couchdb/source/cmd/controller/main.go